app/src/main/java/com/example/tasky/feature_agenda/data/repository/EventRepositoryImpl.kt

```diff
@@ -83,25 +123,24 @@ class EventRepositoryImpl(
             return Resource.Success(localEvent.toEvent())
         }
 
-        val inputData = Data.Builder()
-            .putString("action", "get")
-            .putString("eventId", eventId)
-            .build()
-
-        enqueueWorker(
-            workManager = workManager,
-            inputData = inputData,
-            requestBuilder = OneTimeWorkRequestBuilder<EventWorker>()
-        )
-
-        return Resource.Success()
+        return try {
+            val response = api.getEvent(eventId)
+            val event = response.body()?.toEvent() ?: return Resource.Error(message = "No such event found!", errorType = ErrorType.OTHER)
+
+            val eventEntity = event.toEventEntity()
+            db.eventDao.upsertEvent(eventEntity)
+            val newEvent = db.eventDao.getEventWithAttendees(eventId.toInt()) ?: return Resource.Error(message = "No such event found!", errorType = ErrorType.OTHER)
+
+            Resource.Success(newEvent.toEvent())
+        } catch(e: HttpException) {
+            Resource.Error(message = e.message ?: "Invalid Response", errorType = ErrorType.HTTP)
+        } catch(e: IOException) {
+            Resource.Error(message = e.message ?: "Couldn't reach server", errorType = ErrorType.IO)
+        }
```

Comment:
Ideally, you should first return the locally saved event and then rather emit an update if there is one from the API. In 99% of the cases there won't be an update for this function, since you'll call it when the user taps on an event in their agenda and an update would only exist if the user changed the event in between on a second device. Yet, you always have that extra delay from the API call here which the user doesn't always need to see.